Source/PluginEditor.cpp

```diff
@@ -62,15 +67,21 @@ void DelayPluginAudioProcessorEditor::resized()
 {
     // This is generally where you'll want to lay out the positions of any
     // subcomponents in your editor..
-    auto width  = 100;
-    auto bounds = getLocalBounds().withTop(getLocalBounds()
-            .getCentreY())
-            .withBottom(getHeight()*0.75)
-            .withWidth(width)
-            .withX(getLocalBounds().getCentreX() - (width / 2));
+    FlexBox                  knobBox;
+    knobBox.flexWrap       = FlexBox::Wrap::wrap;
+    knobBox.justifyContent = FlexBox::JustifyContent::center;
+    knobBox.alignContent   = FlexBox::AlignContent::center;
+    knobBox.flexDirection  = FlexBox::Direction::column;
+
+    knobs.add(&delayTimeSlider); // i think the way ive added these objects to the array is causing a crash on shutdown.

```

Comment:
You're adding the sliders to an OwnedArray. The OwnedArray takes unique ownership of things passed to it. The reason it takes pointers to objects is because it expects those objects to have been dynamically allocated on the heap. So when the owned array goes out of scope it will delete each of the objects (via pointers) that it has been given.

In your case you have declared delayTimeSlider and delayDryWetSlider on the stack, you haven't dynamically created them on the heap (via new). So things exploded when OwnedArray tries to delete these at the end of your program.

See usage from the JUCE docs:

> Declare it in the form: OwnedArray<MyObjectClass>
> ..and then add new objects, e.g. myOwnedArray.add (new MyObjectClass());

Given you only have two sliders here I'm not sure using an array is worth it. However, if you did add more, you could use something like a std::unordered_map to store sliders with an ID so you can refer to them later.